src/webapi/document.rs

```diff
@@ -190,6 +192,38 @@ impl Document {
             return @{self}.importNode( @{n.as_ref()}, @{deep} );
         ).unwrap()
     }
+
+    /// Check if the fullscreen API is enabled
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/Document/fullscreenEnabled)
+    // https://fullscreen.spec.whatwg.org/#dom-document-fullscreenenabled
+    pub fn fullscreen_enabled( &self ) -> bool {
+        match js!( return @{self}.fullscreenEnabled; ) {
+            Value::Bool(value) => value,
+            _ => false, // if the variable is not set as a bool, then assume fullscreen is not supported
+        }
+    }
+
+    /// Get the current fullscreen element, or None if there is nothing fullscreen
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/DocumentOrShadowRoot/fullscreenElement)
+    // https://fullscreen.spec.whatwg.org/#dom-document-fullscreenelement
+    pub fn fullscreen_element( &self ) -> Option<Element> {
+        Some(js!( return @{self}.fullscreenElement; )
+            .into_reference()?
+            .downcast::<Element>()?)
+    }
+
+    /// Request the page return from fullscreen mode to a normal state
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/Document/exitFullscreen)
+    // https://developer.mozilla.org/en-US/docs/Web/API/Document/exitFullscreen
+    pub fn exit_fullscreen(&self) -> TypedPromise<(), TypeError> {
```

Comment:
Since this returns a promise it should have `#[cfg(feature = "experimental_features_which_may_break_on_minor_version_bumps")]` attached to it for now.